lib/src/robot/client.dart

```diff
@@ -52,6 +54,18 @@ class RobotClientOptions {
       : dialOptions = DialOptions()..credentials = Credentials.locationSecret(locationSecret);
 }
 
+/// {@category Viam SDK}
+/// Represents a discovery query in the SDK to query for discoverable components.
+class DiscoveryQuery {
+  final String subtype;
+  final String model;
+  final Map<String, dynamic> extra;
+
+  DiscoveryQuery({required this.subtype, required this.model, Map<String, dynamic>? extra}) : extra = extra ?? {};
+
+  Struct get extraStruct => extra.toStruct();
+}
+
```

Comment:
I don't think this class should exist and that people should just use the `DiscoveryQuery` proto directly. This doesn't add any additional functionality or ergonomics to the developer IMO

Reply:
If we don't make this class, how do we import `rpb`'s proto `DiscoveryQuery` in our app? I tried using `import 'package:viam_sdk/viam_sdk.dart';` and it doesn't seem to import proto logic.

Reply:
```
lib/main.dart:53:19: Error: The method 'DiscoveryQuery' isn't defined for the class '_MyHomePageState'.
 - '_MyHomePageState' is from 'package:sean_flutter_app/main.dart' ('lib/main.dart').
Try correcting the name to the name of an existing method, or defining a method named 'DiscoveryQuery'.
```

Reply:
You can import it from `viam_sdk/protos/...` or you can add it as a `typedef` here similar to how we do it for `CloudMetadata` a few lines above (the preferred way IMO)

Reply:
Got it let me check out the latter way

Reply:
I tried the latter way, I had a hard time figuring out how to initialize the `extra` pb struct. Ended up importing `import 'package:viam_sdk/src/utils.dart';` for the map to struct helper logic. This seems like bad practice though since `utils.dart` seems purposefully excluded as an export from `viam_sdk.dart`. Open to ideas to making it easier to use `extra` without the `Struct get extraStruct => extra.toStruct();` utility in the `DiscoveryQuery` first draft.